### ISSUE TITLE
Doesn't work with Android

### ISSUE THREAD

**Issue by RyanMitchellWilson** [yamill/react-native-orientation#230]
My app has orientation disabled so I am using this to find out when the phone was rotated so that I can do specific things on the screen. IOS works perfectly but on android it will not detect my device turning. 
Right here I have the same app running on both android (bottom) and ios (top):
<img width="1107" alt="screen shot 2017-09-05 at 2 57 03 pm" src="https://user-images.githubusercontent.com/11371200/30083146-9d977648-924a-11e7-82fb-1ae36de46400.png">

Basically in this app I need to control when the screen rotates because rotation is turned of by default. As you can see the ios one is responding to the rotation allowing my custom styles to take effect and rotate the screen but on the android side, the events are not firing which is not allowing my custom styles to take effect. Does anyone know how to get this working in android?

**Comment by andrerfneves** [comment]
If all you want is to check whether the device is rotated Landscape/Portrait, why not use the `onLayout` callback on a parent `View` where you can check whether the WIDTH > HEIGHT or vice-versa and update your `styles` accordingly?

Regardless, we'd need to see some code to be able to diagnose what you are doing different that is not allowing for this to work on the Android side of things.

**Comment by RyanMitchellWilson** [comment]
Well if you have orientation set to portrait only for the app onLayout doesn't get triggered when you rotate the device. So I need this library to be able to figure out when the device is turned since the app doesn't rotate on its own. 

I have found a bit of a work around for this calling `Orientation.unlockAllOrientations()` when the modal opens and `Orientation.lockToPortrait()` when the modal closes but this causes the rest of the app to have the orientations unlocked during that time. It would be great if the library could know your rotating without having to unlock all the orientation, just like how the iOS side works.

**Comment by andrerfneves** [comment]
@RyanMitchellWilson one approach would be to `lockToPortrait()` in your application's entry point (some React component). Then you should be able to `unlockAllOrientations()` in any screens that you want to allow for multiple orientations. Whenever you close these screens, you should just `lockToPortrait` once again.

**Comment by RyanMitchellWilson** [comment]
That is essentially what I did only I just unlockAllOrientations when my component shows then I lockToPortrait when I hide my component. Would be nice if I didn't need to unlock the orientation to detect the orientation changing since that makes this library act differently between android and ios so you'd have to write two sets of logic depending on which platform. If they had the same behavior this would be a lot nicer to work with for a cross platform app.

**Comment by andrerfneves** [comment]
We're aware of the discrepancy between the two platforms and are working towards unifying/documenting them. Maintainers are quite busy so PRs are always welcome. Closing this for now.
